src/ffi.rs

```diff
@@ -31,11 +31,14 @@ impl CPath for Path {
         let path = self.as_os_str().as_bytes().iter().copied();
         let mut clean_string = Vec::new();
 
-        for char in path {
-            if char == 0 {
-                clean_string.extend(null_string.clone())
+        for ch in path {
+            if ch == 0 {
+                #[cfg(windows)]
+                clean_string.extend(null_string.clone());
```

Comment:
just curious, why does this need a `clone`?

Reply:
The earlier `encode_wide` encodes the bytes to UTF-16, which doesn't implement `Copy`. I was just getting this code to compile on all platforms so non-windows was complaining that the clone was unneeded since it was a clone of a reference.

Reply:
Yeah I fixed it up a bit, it even messier then before:
https://github.com/daxpedda/sentry-contrib-native/blob/1bbf83ed7b561fad82d05eb07bbdcf71c0f83589/src/ffi.rs#L29-L61

I am considering just throwing this whole thing out and just panic on `0` bytes. Rust does this too: https://doc.rust-lang.org/std/env/fn.var_os.html.

What do you guys think?

Reply:
I would be in favor of that as at drastically simplifies things, IMO.

Reply:
Gonna do that then.